project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module

```diff
@@ -677,15 +682,139 @@ function c4m_user_access_denied() {
+    '#type' => 'textfield',
+    '#title' => t('From'),
+    '#description' => t("From address"),
+    '#size' => 30,
+    '#default_value' => $user->mail,
+    '#disabled' => TRUE,
+  );
+
+  $form['c4m_user_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#description' => t("Enter mail subject."),
+    '#size' => 10,
   );
+
+  $form['c4m_user_body'] = array(
+    '#title' => t('Body'),
+    '#description' => t('Enter mail body'),
+    '#type' => 'text_format',
+    '#format' => 'full_html',
+  );
+
+  $form['c4m_user_checkbox'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Send me a copy'),
+    '#default_value' => FALSE,
+  );
+
+  $form['c4m_user_submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  );
+
+  return $form;
+}
+
+/**
+ * Validate function for contact_form.
+ *
+ * Verifies that email subject and body are not empty.
+ */
+function c4m_user_contact_form_validate($form, $form_state) {
+  $subject = $form_state['values']['c4m_user_subject'];
+  $body = $form_state['values']['c4m_user_body']['value'];
+
+  if (empty($subject)) {
+    form_error($form['c4m_user_subject'], 'Subject field must have a value!');
+  }
+
+  if (empty($body)) {
+    form_error($form['c4m_user_body'], 'Body field must have a value!');
+  }
+}
+
+/**
+ * Submit function for contact_form.
+ *
+ * Generates required parameters and sends an email.
+ * If 'Send copy' checkbox is checked, sends copy mail to running user.
+ */
+function c4m_user_contact_form_submit($form, $form_state) {
+  global $user;
+
+  // Generate and send mail.
+  $account = $form_state['user'];
+  $subject = $form_state['values']['c4m_user_subject'];
+  $body = $form_state['values']['c4m_user_body']['value'];
+  $send_copy = $form_state['values']['c4m_user_checkbox'] == 1;
+
+  _c4m_user_send_mail($account->mail, $user->mail, $subject, $body);
+
+  if ($send_copy) {
+    // If 'Send copy' checkbox is checked, send copy mail.
+    _c4m_user_send_mail($user->mail, $user->mail, "Copy: " . $subject, $body);
+  }
+}
+
+/**
+ * Generates and sends an email.
+ *
+ * @param string $to
+ *   Email destination address.
+ * @param string $from
+ *   Email source address - current user email.
+ * @param string $subject
+ *   Email message subject.
+ * @param string $body
+ *   Email message body.
+ */
+function _c4m_user_send_mail($to, $from, $subject, $body) {
+  $module = 'c4m_user';
+  $key = 'user_contact_mail';
+  $language = language_default();
+  $params = array();
+  $send = FALSE;
+
+  $message = drupal_mail($module, $key, $to, $language, $params, $from, $send);
+  $message['subject'] = $subject;
+  $message['body'] = array();
+  $message['body'][] = $body;
+
+  // Retrieve the responsible implementation for this message.
+  $system = drupal_mail_system($module, $key);
+
+  // Format the message body.
+  $message = $system->format($message);
+
+  // Send e-mail.
+  $message['result'] = $system->mail($message);
```

Comment:
This is not on the issue, so no (the redirect).

You can add a message to indicate success/failure with `drupal_set_message()`